File: JavaScript/8js 1/findAverage_mean.js
/*
|--------------------------------------------------------------------------
| Grasshopper - Array Mean
|--------------------------------------------------------------------------
|
*/

function findAverage(nums) {
   return nums.reduce((a, b) => a + b) / nums.length;
}


const findAverage = (nums) => nums.reduce((pre, val) => pre + val, 0) / nums.length;


function findAverage(nums) {
   let res = 0;
   for (let num of nums) {
      res += num;
   }
   return res / nums.length;
}


console.log(findAverage([1]), 1);
console.log(findAverage([1, 3, 5, 7]), 4);

File: Python/8py 1/find_average_mean.py
"""
|--------------------------------------------------------------------------
| Grasshopper - Array Mean
|--------------------------------------------------------------------------
|
"""

def find_average(nums):
   return sum(nums) / len(nums) if len(nums) != 0 else 0


def find_average(nums):
   return sum(nums) / len(nums) if nums else 0


from numpy import mean
def find_average(nums):
   return mean(nums) if nums else 0


print(find_average([1]), 1)
print(find_average([1, 3, 5, 7]), 4)
print(find_average([-1, 3, 5, -7]), 0)
print(find_average([5, 7, 3, 7]), 5.5)
print(find_average([]), 0)